xwiki-platform-core/xwiki-platform-administration/xwiki-platform-administration-ui/src/main/resources/XWiki/Registration.xml

```diff
@@ -421,6 +421,14 @@
       {{translation key="core.register.invalidUsername"/}}
     #elseif($reg == -11)
       {{translation key="core.register.mailSenderWronglyConfigured"/}}
+    #elseif($reg == -9)
```

Comment:
Out of curiosity, are those magic numeric values explained somewhere?

Reply:
To my knowledge no, they're related to the error numbers returned by `XWiki#createUser` there's quite a few todo in there to improve it...